lib/cli.js

```diff
@@ -57,22 +58,29 @@ exports.run = function () {
         require('source-map-support').install(sourceMapOptions);
     }
 
-    const scripts = internals.traverse(settings.paths, settings);
+    const scripts = await internals.traverse(settings.paths, settings);
+
+    const isNotCjs = (script) => script._type !== 'cjs';
+    if (settings.coverage && scripts.some(isNotCjs)) {
```

Comment:
Perhaps we should have similar checks for `settings.typescript` and `settings.transform`.  I am not sure how common it could be to find projects mixing both ESM and typescript/babel.  It could definitely lead to some confusing issues.  If we'd like to cater to it, another option is to run the tests but output a warning and exit `1` unless it is acknowledged by passing some kind of additional flag.

Reply:
Hmm, yeah transforms, including typescript, also won't work through `import`. It would make sense to reject those as well.

While it is unlikely to combine ESM modules with imports from typescript code, it could happen while migrating. FYI, I already have projects which has commonjs test cases, but the rest of the logic is based on typescript.

Typescript has experimental ESM node.js output support to complicate matters further: https://www.typescriptlang.org/docs/handbook/esm-node.html